GVFS/GVFS.Platform.Mac/ProjFSKext.cs

```diff
@@ -73,5 +69,73 @@ public bool TryPrepareFolderForCallbacks(string folderPath, out string error, ou
 
             return true;
         }
+
+        private ProcessResult Bash(string cmd)
+        {
+            string escapedArgs = cmd.Replace("\"", "\\\"");
+
+            ProcessStartInfo startInfo = new ProcessStartInfo
+            {
+                FileName = "/bin/bash",
+                Arguments = $"-c \"{escapedArgs}\"",
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+                UseShellExecute = false,
+                CreateNoWindow = true,
+            };
+
+            using (Process process = new Process())
+            {
+                process.StartInfo = startInfo;
+                string errors = string.Empty;
+                process.ErrorDataReceived += (sender, args) =>
+                {
+                    if (args.Data != null)
+                    {
+                        errors += args.Data + "\r\n";
+                    }
+                };
+
+                process.Start();
+                process.BeginErrorReadLine();
+                string output = process.StandardOutput.ReadToEnd();
+                process.WaitForExit();
+                return new ProcessResult(output, errors, process.ExitCode);
+            }
+        }
+
+        private bool TryLoadKext(ITracer tracer, string enlistmentRoot, out string error)
+        {
+            Console.WriteLine("Kernel extension not loaded.  Attempting to load...");
+            error = DriverName + " is not loaded. Make sure the driver is loaded and try again.";
+            ProcessResult loadKext = this.Bash("sudo kextutil /Library/Extensions/PrjFSKext.kext");

```

Comment:
Is there a way to get this path other than hard coding?

Reply:
Instead of a path, you can use the bundle identifier using the `-b` flag. (`sudo kextutil -b io.gvfs.PrjFSKext`, at least until we change it for #584.)

Reply:
1. can there be an alternative which does not need `sudo`? `GVFS.Service` would be running elevated, right? Can the service do the kext loading without a `sudo`?
2. can we use the full path to the tool here - like `/sbin/kextload`. Partial paths depend on runtime environment to construct the full path, and may not be the right thing to do while using `sudo`.
3. shouldn't it be `kextload`? rather than `kextutil`?

Reply:
@pmj 
How does it determine where the actual `.kext` file is if we switch to use the bundle identifier?

@alameenshah 
1. We don't have the service on Mac yet.  At least don't think we do.  Correct me if I am wrong.
2. Will you explain this more?  Couldn't someone could just move it?  If we hard code the full path can we ensure it will always be there?  When might it not be the right thing to do using the environment?
3. It was `kextutil` in the `LoadPrjFSKext.sh` so that is why I kept it using `kextutil` even though it is using `kextunload` in that script as well.  I will try `kextload` and make sure it uses the same exit codes.

Reply:
>>Couldn't someone could just move it? If we hard code the full path can we ensure it will always be there?

The path `/sbin/kextload` is protected by [SIP](https://support.apple.com/en-us/HT204899), so it will be there.

>> When might it not be the right thing to do using the environment?

We know the path to the tool that we want to use - `/sbin/kextload`. And by specifying the full path we don't depend on the shell to expand it to what we intended.

Reply:
> 3\. shouldn't it be `kextload`? rather than `kextutil`?

`kextutil` can do anything `kextload` can; if you're showing its output to users, `kextutil` might be better, as it's intended for diagnosing problems.

Reply:
> @pmj
> How does it determine where the actual `.kext` file is if we switch to use the bundle identifier?

It considers all kexts in `/System/Library/Extensions` and al that have passed validation in `/Library/Extensions`.

Reply:
> 1. can there be an alternative which does not need `sudo`? `GVFS.Service` would be running elevated, right? Can the service do the kext loading without a `sudo`?

Incidentally, we do have PrjFSKextLogDaemon running all the time; this could offer an IPC service for attempting to recover from common kext-not-loaded/kext-not-loadable situations.

I should probably mention that it's possible to mark a kext as loadable by non-root users using the [`OSBundleAllowUserLoad`Info.plist key](https://developer.apple.com/library/archive/documentation/Darwin/Conceptual/KEXTConcept/Articles/infoplist_keys.html). However:

> `OSBundleAllowUserLoad`<br>
> The OSBundleAllowUserLoad property allows non-root users to load your kext. Using this property is not recommended.
> 
> I/O Kit drivers should never include this property, because they are loaded by the kernel when they are needed.